### ISSUE TITLE
ufw, iptables xtables-addon

### ISSUE THREAD

**Issue by anoma** [mail-in-a-box/mailinabox#382]
I have been closely monitoring and refining fail2ban fora a future PR to add more elements however it should come as no surprise that most of the IPs fail2ban catches are geoip specific.

At this time i estimate over 95% of all fail2ban triggers are sourced from China.

Given that I went looking for a geoip alternatives and whilst there are no production ready fail2ban options there is a iptables/ufw option:

http://xtables-addons.sourceforge.net/

I have not ran it yet however I see many references to people using it in Ubuntu mailing lists probably because it is available as a package in native ubuntu repo.

A geoip database would need imported regularly but in theory once the groundwork is complete we have an option to add rules will "--src-cc CN" and all the block/rate limit options that come with it.

Would we consider this as an option?

**Comment by JoshData** [comment]
Putting tighter restrictions on a whole country doesn't sound right to me. It wouldn't be fair to users living in China, for instance.

**Comment by anoma** [comment]
I understand that point of view.

However keep in mind we are not necessarily blocking whole countrys although that option would exist. For example we could ufw rate limit countrys that are known problematic users reducing their ability to bruteforce services, send spam whilst allowing legitimate mail through.

It could be argued this is the sysadmin decision and these tools only given them more options. 

Certainly from my point of view blocking large China ISPs from SSH has reduced my attack traffic by VAST amounts. I see no down side on that as I will not be travelling to China any time soon and needing SSH access .

**Comment by JoshData** [comment]
I'm going to close this issue, but feel free to open a new issue if there's a way to address this problem without blocking a whole country by default (and, as I say elsewhere, I don't want to provide any end-user options like this --- everything has to just work by default).

**Comment by roelfrenkema** [comment]
Whzy?

**Comment by ChiefGyk** [comment]
If they need to add a country block prior... I had a better solution for countries or even Tor exit nodes as well as blocking malicious address here https://github.com/ChiefGyk/ipset-assassin 
This project is for people anywhere to be able to accept and send email privately so it makes sense not to exclude it as I did in #875 which is based on the blocklist I mentioned prior. My blocklist works perfectly fine alongside MiaB if you really need to block countries.
